### PR TITLE
feat: Add element select button

### PR DESCRIPTION
**Pull request by SHession**
## What does this change?

This attempts to improve the select-ability of elements by providing a convenient button which will select the element. This should make working with them much simpler, improving the workflows for copying, cutting and deleting elements. It should also improve how intuitive it is for changes to be undone and redone. See the video for an illustration of these effects.

https://user-images.githubusercontent.com/4633246/152768970-1034d24c-9714-48b2-b607-f984eed89d9a.mov

## How to test

This PR can be tested in the sandbox environment using this branch. I have also created a [branch in Composer](https://github.com/guardian/flexible-content/compare/pme-selectable-elements-beta?expand=1) using a beta release to see the finished product. 

## How can we measure success?

It provides a new and intuitive way to interact with elements.

## Have we considered potential risks?

This may causes confusion in Composer during the transition from old to new element types.

## Accessibility

- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [x] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [x] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [x] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [x] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [x] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

### CODE REVIEW

File: src/renderers/react/ElementWrapper.tsx [hunk 1]
```diff
@@ -37,10 +38,25 @@ const Body = styled("div")`
 
 const Panel = styled("div")<{ isSelected: boolean }>`
   background-color: ${({ isSelected }) =>
-    isSelected ? "#b3d7fe" : neutral[97]};
+    isSelected ? "#b3d7fe !important" : neutral[97]};
```
Comment (jonathonherbert) [inline src/renderers/react/ElementWrapper.tsx:41]: Haha. It works – but it's brutal!

I'm sure you've tried lots of things – is there any way we can avoid this? Appreciate we don't necessarily want to be passing selection state around.

My first thought is a transparent coloured overlay and some conditional styling for selection, but you may have already been down this road! 

We could also use the [component selector](https://styled-components.com/docs/advanced#referring-to-other-components) pattern for CSS precedence.

Not a blocker, but `!important` does make me itch a bit 😅

File: src/plugin/plugin.ts [hunk 2]
```diff
@@ -83,6 +84,12 @@ export const createPlugin = <
             ...node.attrs,
             [elementSelectedNodeAttr]: isCurrentlySelected,
           });
+
+          if (selection instanceof NodeSelection) {
```
Comment (jonathonherbert) [inline src/plugin/plugin.ts:88]: This will happen for every descendant node – IIUC we just want it to happen once, if any element node has been modified.

**Review by jonathonherbert — COMMENTED** [review]
Really like the icon and the behaviour. A few comments on code to resolve, one non-blocking styling suggestion.

File: src/renderers/react/ElementWrapper.tsx (hunk 1 above)
Comment (SHession) [reply src/renderers/react/ElementWrapper.tsx]: Switched to using an overlay as per the suggestion. In my opinion it looks much better.

File: src/renderers/react/ElementWrapper.tsx [hunk 3]
```diff
@@ -201,12 +238,16 @@ export const ElementWrapper: React.FunctionComponent<Props> = ({
...
             }}
+            aria-label="Delete element"
           >
             <SvgBin />
             {closeClickedOnce && <Tooltip>Click again to confirm</Tooltip>}
           </SeriousButton>
         </LeftActions>
-        <Panel isSelected={isSelected}>{children}</Panel>
+        <Panel isSelected={isSelected}>
+          <Overlay isSelected={isSelected} />
```
[the remark this answers is not in the document]
Comment (SHession) [reply src/renderers/react/ElementWrapper.tsx]: Good suggestion, it is slightly nicer. Although I believe `pointer-events: none;` meant there were no issues with element interaction.

**Pull request merged**